packages/core-mobile/app/hooks/earn/useGetFeeState.ts

```diff
@@ -0,0 +1,34 @@
+import { pvm } from '@avalabs/avalanchejs'
+import { useAvalancheXpProvider } from 'hooks/networks/networkProviderHooks'
+import { useCallback, useEffect, useState } from 'react'
+import Logger from 'utils/Logger'
+
+export const useGetFeeState = (): {
+  getFeeState: (gasPrice?: bigint) => pvm.FeeState | undefined
+} => {
+  const xpProvider = useAvalancheXpProvider()
+  const [feeState, setFeeState] = useState<pvm.FeeState | undefined>(undefined)
```

Comment:
can we use react query to cache the fee state? feel like it's more efficient that way and we can also add interval update if necessary.